examples/simple_publisher.c

```diff
@@ -76,7 +76,12 @@ int main(int argc, const char *argv[])
     uint8_t sendbuf[2048]; /* sendbuf should be large enough to hold multiple whole mqtt messages */
     uint8_t recvbuf[1024]; /* recvbuf should be large enough any whole mqtt message expected to be received */
     mqtt_init(&client, sockfd, sendbuf, sizeof(sendbuf), recvbuf, sizeof(recvbuf), publish_callback);
-    mqtt_connect(&client, "publishing_client", NULL, NULL, 0, NULL, NULL, 0, 400);
+    /* Create an anonymous session */
+    const char* client_id = NULL;
```

Comment:
I don't think the simple example should be of an anonymous client.

Reply:
In that case it should use a random(ish) name.
The reason I implemented this is because I inhibited a session from someone else testing.
Also, if 2 concurrent users connect with the same client name, one will be dropped.

Reply:
Oh I see what you mean. That's a good point.